jwst/assign_wcs/assign_wcs_step.py

```diff
@@ -83,5 +90,45 @@ def process(self, input, *args, **kwargs):
             slit_y_range = [self.slit_y_low, self.slit_y_high]
             result = load_wcs(input_model, reference_file_names, slit_y_range)
 
+            if self.sip_approx:
+                # check if its imaging mode
+                if result.meta.exposure.type.lower() in IMAGING_TYPES:
```

Comment:
From what I see, these `if`s can also be joined with an `and`:
```suggestion
            if result.meta.exposure.type.lower() in IMAGING_TYPES and self.sip_approx:
```
and then decrease indentation of the rest of the block.

Reply:
Even better:
```python
if not (result.meta.exposure.type.lower() in IMAGING_TYPES and self.sip_approx):
    return result
```
and then de-dent by 8 spaces!